client/clientset/fake/extensions.go

```diff
@@ -31,11 +32,11 @@ func NewFakeExtensionClient(objects ...runtime.Object) *FakeExtensionClient {
 	return &FakeExtensionClient{&fakePtr}
 }
 
-func (m *FakeExtensionClient) DatabaseSnapshots(ns string) client.DatabaseSnapshotInterface {
+func (m *FakeExtensionClient) DatabaseSnapshot(ns string) client.DatabaseSnapshotInterface {
```

Comment:
I would recommend keep the names plural here and fix the interface.
@TamalSaha @aerokite what do you think?